### PR TITLE
GitHub Action improvements, including coverage tests!

### PR DESCRIPTION
**Pull request by rspier**
- Various cleanups of obsolete things from the workflow.
- Caching so we don't have to install modules each time (saves 2 minutes)
- Run Devel::Cover coverage tests and push results to Coveralls if the want-coverage label is applied.
- Explicitly use MariaDB, instead of whatever Debian's default is.

### CODE REVIEW

File: .github/workflows/perl-tests.yml [hunk 1]
```diff
@@ -30,13 +30,24 @@ jobs:
...
+            /usr/local/lib/perl5/site_perl
+          key: ${{ runner.os }}-${{ steps.get-perl-version.outputs.version }}-${{ hashFiles('Makefile.PL') }}
       - name: Install prereqs (cpan)
         # This could probably be made more efficient by looking at what it's
         # installing via cpanm that could, instead, be installed from apt.  I
         # may do that later, but for now, it's fine! -- rjbs, 2023-01-07
         run: cpanm --notest --installdeps .
       - name: Run the tests
-        if: !(contains(github.event.pull_request.labels.*.name, 'want-coverage'))
+        if: "!(contains(github.event.pull_request.labels.*.name, 'want-coverage'))"
```
Comment (rjbs) [inline .github/workflows/perl-tests.yml:54]: does this belong in its own commit?  I think yes: not a showstopper but also easy to change

**Review by rjbs — APPROVED** [review]
Looks good.  Took me a bit to make sure I understood it.  I noted one very very small nit in git discipline.

File: .github/workflows/perl-tests.yml (hunk 1 above)
Comment (rspier) [reply .github/workflows/perl-tests.yml:54]: Good catch.  It actually belongs in the first commit, not this one.

Comment (rspier) [reply .github/workflows/perl-tests.yml:54]: (i.e. it was a yaml format error without the quotes, and apparently I didn't add it to the right one the first time.  It is now part of the first commit that adds coverage.)
